src/Microsoft.Graph.Core/Tasks/PageIterator.cs

```diff
@@ -230,14 +233,50 @@ public async Task ResumeAsync()
         /// <returns>The task object that represents the results of this asynchronous operation.</returns>
-        /// <exception cref="Microsoft.CSharp.RuntimeBinder.RuntimeBinderException">Thrown when a base CollectionPage that does not implement NextPageRequest
-        /// is provided to the PageIterator</exception>
         /// <exception cref="Microsoft.Graph.ServiceException">Thrown when the service encounters an error with
         /// a request.</exception>
         public async Task ResumeAsync(CancellationToken token)
         {
             await IterateAsync(token);
         }
+
+        /// <summary>
+        /// Helper method to extract the collection rom an <see cref="IParsable"/> instance.
+        /// </summary>
+        /// <param name="parsableCollection">The <see cref="IParsable"/> to extract the collection from</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentException">Thrown when the object doesn't contain a collection inside it</exception>
+        private static List<TEntity> ExtractEntityListFromParsable(IParsable parsableCollection)
+        {
+            return parsableCollection.GetType().GetProperty("Value").GetValue(parsableCollection, null) as List<TEntity> ?? throw new ArgumentException("The Parsable does not contain a collection property");
+        }
+
+        /// <summary>
+        /// Helper method to execute the the request to get back the next page
+        /// </summary>
+        /// <param name="nextPageRequestInformation">The <see cref="RequestInformation"/> of the next page request</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use to cancel requests</param>
+        /// <returns></returns>
+        private async Task<IParsable> GetNextPageAsync(RequestInformation nextPageRequestInformation, CancellationToken cancellationToken)
+        {
+            // Use reflection to invoke the "requestAdapter.SendAsync" method so as to be able to be able to pass the generic type arguments
+            // neededed for handling the response.
+            var method = _client.RequestAdapter.GetType().GetMethod("SendAsync").MakeGenericMethod(_currentPage.GetType());
+            var sendTask = (Task)method.Invoke(_client.RequestAdapter, new object[] { nextPageRequestInformation, null, cancellationToken });
+            await sendTask.ConfigureAwait(false);
+            // return the result from the task.
+            return (IParsable)sendTask.GetType().GetProperty("Result").GetValue(sendTask);
+        }
+
+        /// <summary>
+        /// Helper method to extract the nextLink property from an <see cref="IParsable"/> instance.
+        /// </summary>
+        /// <param name="parsableCollection">The <see cref="IParsable"/> to extract the nextLink from</param>
+        /// <returns></returns>
+        private static string ExtractNextLinkFromParsable(IParsable parsableCollection)
```

Comment:
make next link a parameter so you can reuse that method for delta link and affect delta link, I don't thing it's ever affected to anything other than string.empty now.

Reply:
resolved in latest update.